code/game/objects/items/explosives/mine.dm

```diff
@@ -168,38 +227,56 @@
 
 
 /obj/item/explosive/mine/proc/set_tripwire()
-	if(!active && !tripwire)
+	if(!active && !tripwire && use_tripwire)
 		var/tripwire_loc = get_turf(get_step(loc, dir))
 		tripwire = new(tripwire_loc)
 		tripwire.linked_claymore = src
-		active = TRUE
+	active = TRUE
+
 
 
 //Mine can also be triggered if you "cross right in front of it" (same tile)
-/obj/item/explosive/mine/Crossed(atom/A)
+/obj/item/explosive/mine/Crossed(atom/victim)
 	..()
-	if(isliving(A))
-		var/mob/living/L = A
-		if(!L.stat == DEAD)//so dragged corpses don't trigger mines.
+	if(isliving(victim))
+		var/mob/living/unfortunate_soul = victim
+		if(heavy_trigger && buried)
+			unfortunate_soul.visible_message(SPAN_DANGER("[icon2html(src, viewers(src))] [name] clicks as [unfortunate_soul] moves in front of it."), \
```

Comment:
this is gonna cause double messages with the try_to_prime proc

Reply:
reworked section entirely to call a new heavy trigger proc that just does that and removed that code from crossed()